test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1

```diff
@@ -0,0 +1,109 @@
+
+$openCoverPath = "$outputBaseFolder\OpenCover"
+$testPath = "$outputBaseFolder\tests"
+$psBinPath = "$outputBaseFolder\PSCodeCoverage"
+$openCoverTargetDirectory = "$outputBaseFolder\OpenCoverToolset"
+$outputLog = "$outputBaseFolder\CodeCoverageOutput.xml"
+$psCodeZip = "$outputBaseFolder\PSCode.zip"
+$psCodePath = "$outputBaseFolder\PSCode"
+
+try
+{
+    "Starting downloads." | Tee-Object -FilePath "$PSScriptRoot\logs.txt" -Append
+
+    Invoke-WebRequest -uri $codeCoverageZip -outfile "$outputBaseFolder\PSCodeCoverage.zip"
+    Invoke-WebRequest $testContentZip -outfile "$outputBaseFolder\tests.zip"
+    Invoke-WebRequest $openCoverZip -outfile "$outputBaseFolder\OpenCover.zip"
+
+    "Downloads complete." | Tee-Object -FilePath "$PSScriptRoot\logs.txt" -Append
+
+    "Starting expansion." | Tee-Object -FilePath "$PSScriptRoot\logs.txt" -Append
+
+    Expand-Archive -path "$outputBaseFolder\PSCodeCoverage.zip" -destinationpath "$psBinPath"
+    Expand-Archive -path "$outputBaseFolder\tests.zip" -destinationpath $testPath
+    Expand-Archive -path "$outputBaseFolder\OpenCover.zip" -destinationpath $openCoverPath
+
+    ## Download Coveralls.net uploader
+    $coverallsToolsUrl = 'https://github.com/csMACnz/coveralls.net/releases/download/0.7.0/coveralls.net.0.7.0.nupkg'
+    $coverallsPath = "$outputBaseFolder\coveralls"
+
+    ## Saving the nupkg as zip so we can expand it.
+    Invoke-WebRequest $coverallsToolsUrl -outfile "$outputBaseFolder\coveralls.zip"
+    Expand-Archive -Path "$outputBaseFolder\coveralls.zip" -DestinationPath $coverallsPath
+
+    "Expansion complete." | Tee-Object -FilePath "$PSScriptRoot\logs.txt" -Append
+
+    Import-Module "$openCoverPath\OpenCover" -force
+    Install-OpenCover -TargetDirectory $openCoverTargetDirectory -force
+    "OpenCover installed." | Tee-Object -FilePath "$PSScriptRoot\logs.txt" -Append
+
+    "TestDirectory : $testPath" | Tee-Object -FilePath "$PSScriptRoot\logs.txt" -Append
+    "openCoverPath : $openCoverTargetDirectory\OpenCover" | Tee-Object -FilePath "$PSScriptRoot\logs.txt" -Append
+    "psbinpath : $psBinPath" | Tee-Object -FilePath "$PSScriptRoot\logs.txt" -Append
+
+    $openCoverParams = @{outputlog = $outputLog;
+                         TestDirectory = $testPath;
+                         OpenCoverPath = "$openCoverTargetDirectory\OpenCover";
+                         PowerShellExeDirectory = "$psBinPath\publish"
+                        }
+
+    $openCoverParams | Tee-Object -FilePath "$PSScriptRoot\logs.txt" -Append
+
+    "Starting test run." | Tee-Object -FilePath "$PSScriptRoot\logs.txt" -Append
+
+    Invoke-OpenCover @openCoverParams
+
+    if(Test-Path $outputLog)
+    {
+        get-childitem $outputLog | Tee-Object -FilePath "$PSScriptRoot\logs.txt" -Append
+    }
+
+    "Test run done.!!" | Tee-Object -FilePath "$PSScriptRoot\logs.txt" -Append
+
+    $powershellVersionString = (Get-Content "$psBinPath\publish\powershell.version")
```

Comment:
I expect this file to go away at some point - it would be better to get the string from PowerShell itself, e.g.

```powershell
$powershellVersionString = & $psBinPath\publish\powershell.exe -noprofile -command '$PSVersionTable.GitCommitId'
```